app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -4519,9 +4538,13 @@ private void setupVideoSurface() {
             surfaceHolderCallback = new SurfaceHolderCallback(context, simpleExoPlayer);
             binding.surfaceView.getHolder().addCallback(surfaceHolderCallback);
             final Surface surface = binding.surfaceView.getHolder().getSurface();
-            // initially set the surface manually otherwise
-            // onRenderedFirstFrame() will not be called
-            simpleExoPlayer.setVideoSurface(surface);
+            // ensure player is using an unreleased surface, which the surfaceView might not be
+            // when starting playback on background or during player switching
+            if (surface.isValid()) {
+                // initially set the surface manually otherwise
+                // onRenderedFirstFrame() will not be called
+                simpleExoPlayer.setVideoSurface(surface);
```

Comment:
Could this be replaced by `simpleExoPlayer.setVideoSurfaceHolder(binding.surfaceView.getHolder());`, or would there be issues? After some testing in #8170 I couldn't notice any difference.

Reply:
Absolutely, that is the better solution. `setVideoSurfaceHolder` is doing the same null checks here: https://github.com/google/ExoPlayer/blob/release-v2/library/core/src/main/java/com/google/android/exoplayer2/ExoPlayerImpl.java#L1262